include/session/config/groups/members.hpp

```diff
@@ -287,11 +314,14 @@ struct member {
 
   private:
     friend class Members;
-    void load(const dict& info_dict);
+    void load(const dict& info_dict, const dict& extra_dict);
 };
 
 class Members : public ConfigBase {
 
+  private:
+    std::optional<std::unordered_set<std::string>> pending_send_ids;
```

Comment:
```suggestion
    std::optional<std::unordered_set<std::string>> pending_send_ids;
```
I don't think having this be `optional` is useful; what optional gives you is an extra "no data" state, which is useful if you want to distinguish between "this is not set at all" and "this is empty".  But here it seems like empty is perfectly fine, and would simply things.

Reply:
I wasn't sure if an empty `std::unordered_set<std::string>` would take up memory (though now that I write that, it should be negligible even if it does so is probably just an unnecessary optimisation)